contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -508,14 +488,19 @@ abstract contract FunctionsSubscriptions is IFunctionsSubscriptions, ERC677Recei
   /**
    * @inheritdoc IFunctionsSubscriptions
    */
-  function timeoutRequests(bytes32[] calldata requestIdsToTimeout) external override {
+  function timeoutRequests(IFunctionsRequest.Commitment[] calldata requestsToTimeoutByCommitment) external override {
     _whenNotPaused();
-    for (uint256 i = 0; i < requestIdsToTimeout.length; ++i) {
-      bytes32 requestId = requestIdsToTimeout[i];
-      Commitment memory request = s_requestCommitments[requestId];
-      uint64 subscriptionId = request.subscriptionId;
+    for (uint256 i = 0; i < requestsToTimeoutByCommitment.length; ++i) {
+      IFunctionsRequest.Commitment memory request = requestsToTimeoutByCommitment[i];
+      bytes32 requestId = request.requestId;
+
+      // Check that request ID is valid
+      if (keccak256(abi.encode(request)) != s_requestCommitments[requestId]) {
+        revert InvalidCalldata();
+      }
 
       // Check that the message sender is the subscription owner
+      uint64 subscriptionId = request.subscriptionId;
       _isValidSubscription(subscriptionId);
       address owner = s_subscriptions[subscriptionId].owner;
       if (msg.sender != owner) {
```

Comment:
I thought we agreed anyone should be able to time out requests (or at the very least CLL should be able to): https://chainlink-core.slack.com/archives/C042Z3736JD/p1690914168586579